### ISSUE TITLE
Amplify Pull in a Bitbucket Pipeline

### ISSUE THREAD

**Issue by kimfucious** [aws-amplify/amplify-cli#3601]
**Which Category is your question related to?**
amplify-cli

**Amplify CLI Version** 
4.13.1

**What AWS Services are you utilizing?**
Almost everything :)

I'm not committing the `aws-exports.js` file to my repo, which causes the app to crash at build in a Bitbucket Pipeline.

I'd like to run Amplify pull in a "non-interactive" way, supplying credentials/options with the repo's environment variables.  I've read #3292, but I can't frankly tell if there's a solution there or not.

Further, this repo is for an app that is using another app's back-end.  there's been no `amplify init` run.  So I really only need the `aws-exports.js" file (which I'm thinking about adding to the repo, and making the repo private, if that's the only route).   

Any advice is always appreciated 🌮 !

**Comment by UnleashedMind** [comment]
Try our Amplify Console service, you can connect it to the Bitbucket repo, it will provide the CICD for you. 

https://us-west-2.console.aws.amazon.com/amplify/home?region=us-west-2#/

**Comment by kaustavghosh06** [comment]
@kimfucious Closing this issue based on @UnleashedMind's response. Please let us know if you need further assistance or have more questions.

**Comment by kimfucious** [comment]
Thanks, @UnleashedMind for the docs.  Sorry for the delay in response, I got distracted by other things.  That link on headless pull were the docs I was looking for.

What I'm actually trying to do here is use CI/CD to run `expo publish` on a React Native app.  I got the idea from [here](https://blog.expo.io/setting-up-expo-and-bitbucket-pipelines-8995ef036a18).

I'll try that out.  Also, ignoring the current issues with Amplify Console's inability to create new environments when using auth configured with social auth providers and connecting to a public repo with `team-provider-info.json` excluded, I think I'll take a look at modifying the `amplify.yml` file and see if I can get Amplify Console to do something similar to what the aforementioned article describes.  

It's a bit different, because `expo publish` puts the app bundles in S3, I believe, and their mobile client app, must use an API to pull the bundles down to the phone.  Frankly, I don't know enough yet to talk about this intelligently nor how "Amplify does React Native".  

To clarify a bit further:
- My current process is manual.
- I've got NPM scripts for dev, staging, and prod.  These will first copy specific versions of `app.json` ( an expo related file) and `aws-config.js` (this is a copy of aws-exports.js) for the respective environments, and then run `expo publish`.
- I'm testing the iOS app in Test Flight, and the app gets updated "OTA" when `expo publish` is run the the `prod` versions of `app.json` and `aws-config.js` 

That's all probably more than you care/wanted to know, but I thought I'd share what I'm trying to accomplish, with the hopes of perhaps gleaning some valuable insight from you fine folks at Amplify!

Any docs toward automating this on AWS would be most appreciated as well.
